File: GigRadio/Models/SoundCloud/SoundCloudTrack.m
//
//  SoundCloudTrack.m
//  GigRadio
//
//  Created by Michael Forrest on 16/08/2014.
//  Copyright (c) 2014 Good To Hear. All rights reserved.
//

#import "SoundCloudTrack.h"
#import "SoundCloudConfiguration.h"
@implementation SoundCloudTrack
+(NSDictionary *)defaultPropertyValues{
    return @{
             @"kind":@"",
             @"id": @0,
             @"duration":@0,
             @"state":@"",
             @"sharing":@"",
             @"tag_list":@"",
             @"permalink":@"",
             @"streamable":@NO,
             @"downloadable":@NO,
             @"purchase_url":@"",
             @"label_id":@0,
             @"purchase_title":@"",
             @"genre":@"",
             @"title":@"",
             @"description":@"",
             @"trackDescription":@"",
             @"label_name":@"",
             @"track_type":@"",
             @"uri":@"",
             @"permalink_url":@"",
             @"artwork_url":@"",
             @"waveform_url":@"",
             @"stream_url":@"",
             @"playback_count":@0,
             @"download_count":@0,
             @"favoritings_count":@0,
             @"comment_count":@0
             };
}


-(NSURL *)playbackURL{
    return [NSURL URLWithString:[self.stream_url stringByAppendingFormat:@"?client_id=%@",[SoundCloudConfiguration clientId]]];
}
@end

File: GigRadio/Models/SoundCloud/SoundCloudTrack.swift
//
//  SoundCloudTrack.swift
//  GigRadio
//
//  Created by Michael Forrest on 17/05/2015.
//  Copyright (c) 2015 Good To Hear. All rights reserved.
//

import UIKit
import RealmSwift

class SoundCloudTrack: Object {
    dynamic var id = 0
    dynamic var kind = ""
    dynamic var createdAt = NSDate()
    dynamic var userId  = 0
    dynamic var duration = 0
    dynamic var state = ""
    dynamic var sharing = ""
    dynamic var tagList = ""
    dynamic var permalink = ""
    dynamic var streamable = false
    dynamic var downloadable = false
    dynamic var purchaseUrl = ""
    dynamic var labelId = 0
    dynamic var purchaseTitle = ""
    dynamic var genre = ""
    dynamic var title = ""
    dynamic var descriptionText = ""
    dynamic var labelName = ""
    dynamic var trackType = ""
    dynamic var uri = ""
    dynamic var permalinkUrl = ""
    dynamic var artworkUrl = ""
    dynamic var waveformUrl = ""
    dynamic var streamUrl = ""
    dynamic var playbackCount = 0
    dynamic var downloadCount = 0
    dynamic var favoritingsCount = 0
    dynamic var commentCount = 0
    
    func playbackURL()->NSURL{
        return NSURL(string: "\(self.streamUrl)?client_id=\(SoundCloudConfiguration().clientId)")!
    }

}
